### PR TITLE
[Accessibility] Focus coupon input if it has an error

### LINKED ISSUE #37602: Accessibility Improvements ‣ Cart | Checkout ‣ Errors and Feedback
As originally noted [over here](https://github.com/woocommerce/woocommerce/issues/24185#issuecomment-779227823), within the cart and checkout pages any errors that display are not programmatically associated with, or within proximity of, their associated fields.

<div align="center">error-association</div>

The potential impact is that users of screen readers will have a difficult time understanding why a form is being rejected, which fields are in error, and how to remedy them. Proposals for us to investigate:

- Ensure `aria-invalid="true"` is on invalid form fields and updated to a`aria-invalid="false"` or removed for valid fields.
- Ensure error messages are nested inside of the element.
- Ensure errors that can't be nested with the element are programmatically associated with the input. Add `aria-describedby` to the input element and point to the ID of the error message.
- Ensure focus moves to the first field that has an error.

This is a relatively large task and in respect of changing where errors display, may require some design input.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #37602 .

In order to create small PRs, the issue above will be fixed along two PRs.
1. This PR is dedicated to making the coupon input accessible when there is an error after sending it on:
    - Classic Cart page.
    - Block-based Cart page.
    - Classic Checkout page.
    - Block-based Checkout page.
2.  [[Accessibility] Improve checkout error messages](https://github.com/woocommerce/woocommerce/pull/49094)

The issue associated with this PR reports more accessibility problems, but most of them have already been fixed by other PRs. Such as:
- [x] Add `aria-invalid="true"` to invalid form fields and update to `aria-invalid="false"` or remove for valid fields.
- [x] Add error messages nested inside the form field.
- [x] On the checkout page, focus the first field with an error.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Note**: I noticed that when the Storefront theme is installed, these CSS files aren't loaded on the classic Cart and Checkout pages:
  - `plugins/woocommerce/client/legacy/css/woocommerce-smallscreen.scss`
  - `plugins/woocommerce/client/legacy/css/woocommerce.scss`
This means that the coupon inline error style will be different depending on the theme installed.

1. On both the Cart and Checkout pages, add a coupon code that doesn't exist.
3. Verify the error message is added below the input. "Coupon "..." does not exist!"
4. Verify the focus is moved to the coupon input.
5. Verify the error message disappears after moving the focus away from the input or as soon as you type any new value to the input.
6. Repeat this process on both the classic and the block-based pages.
7. Test the classic cart against the following themes (check [this comment](https://github.com/woocommerce/woocommerce/pull/48738#pullrequestreview-2204533147) for more context):
    - Twenty Seventeen
    - Twenty Nineteen
    - Twenty Twenty
    - Twenty Twenty-One
    - Twenty Twenty-Two
    - Twenty Twenty-Three
    - Twenty Twenty-Four

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by Manussakis** [comment]
@dkotter This PR is ready for code review.

**Review by dkotter — APPROVED** [review]

**Comment by dkotter** [comment]
@qasumitbagthariya @ankitguptaindia This is ready for QA

**Review by mikejolley — COMMENTED** [review]
This tested well and the code looks good. #37602 however reports the issue with the classic cart experience. This PR only fixes the issue in the block based checkout experience.

If we are fixing classic in a different PR, we should leave #37602 open. Unless it is tackled in this PR instead.

**Comment by Manussakis** [comment]
> This tested well and the code looks good. #37602 however reports the issue with the classic cart experience. This PR only fixes the issue in the block based checkout experience.
> 
> If we are fixing classic in a different PR, we should leave #37602 open. Unless it is tackled in this PR instead.

Good catch @mikejolley !
I will move the issue to the "In Progress" column and tackle the classic cart experience.
Thanks!

**Comment by Manussakis** [comment]
@dkotter This task is ready for code review.
Note that this is the first out of two PRs needed to close the #37602 issue. Please, check the PR description for more context

**Review by dkotter — APPROVED** [review]
@qasumitbagthariya @ankitguptaindia This is ready for QA

**Comment by Manussakis** [comment]
@dkotter FYI – I have made a small change to improve the HTML semantics of the error message.
[Improve coupon error message semantics](https://github.com/woocommerce/woocommerce/pull/48738/commits/0311453e5e8ce0814f8d7d4ac43227c8ccb76162)

**Review by dkotter — APPROVED** [review]

**Comment by vikrampm1** [comment]
@qasumitbagthariya @ankitguptaindia This is ready for QA with the additional changes

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.5.5
* Theme: Main News Version: 1.0.1
* WooCommerce - Dev version with this fix specific branch.  
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 126
* OS: macOS Sonoma 14.5
* Git Branch: Manussakis:fix/37602


</Details>


**Test Results -** 

Changes work as expected.  ✅ 

Tested Cases with Classic and Block Checkout: 

- Focus when the customer enters a correct coupon code. ✅ 
- Focus when the customer enters an incorrect coupon code.✅ 
- When there is more than one coupon on the cart and checkout pages. ✅ 
- When the user enters a correct coupon code after entering an incorrect coupon code. ✅ 
- When the user enters an incorrect coupon code multiple times. ✅ 
- When the user uses an expired coupon code. ✅ 
- When the user uses the same coupon code again. ✅ 
- When a customer removes a coupon code after it has been applied. ✅ 
- When a customer applies multiple coupons sequentially (one after the other). ✅ 


 

**Functional Demo /  Screencast -*


**Classic Cart and Checkout Page:** 

https://github.com/woocommerce/woocommerce/assets/6384299/2758a56c-aa55-4f3f-97c9-6a67c0482ac7


**Block-based Cart Page and Checkout Page:** 


https://github.com/woocommerce/woocommerce/assets/6384299/19359ee9-05a2-4be8-9be6-dabf76a7a32b



**Next Step-** Ready for Code Review.

**Comment by ankitguptaindia** [comment]
@Manussakis This PR is ready for Woo review, please update the title now. Thank you! 


>This PR closes part of the https://github.com/woocommerce/woocommerce/issues/37602 issue. 

Please also link this PR with the issue https://github.com/woocommerce/woocommerce/issues/37602 so it will be visible in the project board view. TY.

**Comment by vikrampm1** [comment]
@mikejolley this PR is ready for your re-review.

**Comment by mikejolley** [comment]
@Manussakis @vikrampm1 Could you add additional instructions for classic cart? I'm trying this out and after submitting the coupon the content is reloaded and nothing has focus. Could it be because I'm using shortcode cart on a block based theme? Notices may have different styles under those circumstances.

**Comment by Manussakis** [comment]
@mikejolley I was able to reproduce the issue you flagged. Block based themes with the shortcode cart are not getting the focus on the coupon input if there is an error.
I'm moving the issue to "In progress" again.

Thanks for catching that.

**Comment by Manussakis** [comment]
@dkotter I have addressed the issue identified by @mikejolley on [this comment](https://github.com/woocommerce/woocommerce/pull/48738#issuecomment-2239298830).
Please re-review the PR and pass it to @ankitguptaindia for a second round of QA if no fixes are needed 🙂.

File: plugins/woocommerce/client/legacy/js/frontend/cart.js [hunk 1]
```diff
@@ -135,12 +135,28 @@ jQuery( function ( $ ) {
...
+			var $old_coupon_error_msg = $( '#coupon_code' )
+					.closest( '.coupon' )
+					.find( '.coupon-error-notice' );
 
 			$( '.woocommerce-cart-form' ).replaceWith( $new_form );
 			$( '.woocommerce-cart-form' )
 				.find( ':input[name="update_cart"]' )
 				.prop( 'disabled', true );
 
+			if ( preserve_notices && $old_coupon_error_msg.length > 0) {
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/cart.js]:
```suggestion
			if ( preserve_notices && $old_coupon_error_msg.length > 0 ) {
```

File: plugins/woocommerce/client/legacy/js/frontend/checkout.js [hunk 2]
```diff
@@ -604,16 +604,45 @@ jQuery( function( $ ) {
...
 		show_coupon_form: function() {
 			$( '.checkout_coupon' ).slideToggle( 400, function() {
 				$( '.checkout_coupon' ).find( ':input:eq(0)' ).trigger( 'focus' );
 			});
 			return false;
 		},
-		submit: function() {
-			var $form = $( this );
+		show_coupon_error: function( html_element, $target ) {
+			if ($target.length === 0) {
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/checkout.js]:
```suggestion
			if ( $target.length === 0 ) {
```

File: plugins/woocommerce/client/legacy/js/frontend/checkout.js [hunk 3]
```diff
@@ -637,18 +666,28 @@ jQuery( function( $ ) {
...
 					$( '.woocommerce-error, .woocommerce-message, .is-error, .is-success' ).remove();
 					$form.removeClass( 'processing' ).unblock();
 
-					if ( code ) {
-						$form.before( code );
-						$form.slideUp();
+					if ( response ) {
+						// We only want to show coupon notices if they are not errors.
+						// Coupon errors are shown under the input.
+						if ( response.indexOf('woocommerce-error') === -1 && response.indexOf('is-error') === -1) {
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/checkout.js]:
```suggestion
						if ( response.indexOf('woocommerce-error') === -1 && response.indexOf('is-error') === -1 ) {
```

File: plugins/woocommerce/client/legacy/js/frontend/checkout.js [hunk 4]
```diff
@@ -637,18 +666,28 @@ jQuery( function( $ ) {
...
+							$form.slideUp(400, function() {
+								$form.before( response );
+							});
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/checkout.js]:
```suggestion
							$form.slideUp( 400, function() {
								$form.before( response );
							} );
```

**Review by dkotter — APPROVED** [review]
@ankitguptaindia This is ready for another round of QA

**Comment by ankitguptaindia** [comment]
**Point-1:**

**Problem Description:** When a customer enters an invalid coupon code, an error message appears in the coupon field. If the customer then tries to interact with another element on the page (such as shipping options or payment option radio buttons), they need to click twice for selecting another option.

This issue occurs only on the Cart Page. The checkout page works fine and does not exhibit this issue.

Testing Details:
- Tested on both shortcode cart and block cart.
- Tested with TT4 theme and non-WC theme. 


Please watch below videos: 

**Classic Cart on TT4** 

https://github.com/user-attachments/assets/9c079d41-e562-4f55-8904-d07045240945


**Block Cart on TT4** 

https://github.com/user-attachments/assets/49e4ba5c-2119-492a-99da-d5fcde1a0235




**Point-2:**

**Problem Description:** When using the classic checkout on a non-block-based theme, the invalid coupon code is automatically cleared. However, with the TT4 theme and block checkout, the invalid coupon code remains in the field. Is this expected behaviour? 



https://github.com/user-attachments/assets/fc3968c9-6dcd-4e77-a3e2-73ed355eb349


_Note: Please let me know if the above concerns are not related to current PR and need to be reported as new issue._

**Comment by Manussakis** [comment]
@ankitguptaindia Thanks for testing it again.

**Point-1**
This is not an issue introduced by this PR. You can reproduce it on the trunk branch by:
- Applying an invalid coupon code.
- Clicking on the input after the error message appears.
- Trying to interact with an element below the input.

This is because the coupon input is not required, so the error message disappears once the focus moves away from it, which happens as soon as you click on any element of the page. When the error message disappears, the height of its container decreases making all the elements below it change their position. That's why the first click doesn't take any effect, because the target changes its initial place.

This issue is more evident now because the focus is automatically moved to the coupon input if there is an error.

I suggest creating a new issue to tackle the bug. I did a quick investigation and noticed that the coupon input uses the same component as several elements on the checkout form (`<ValidatedTextInput>`). I believe it's worth it to centralize the discussion on a dedicated issue.

@mikejolley any thoughts on creating a new issue to deal with the **Point-1** above?

**Point-2**
Good catch! Let's keep consistency and preserve the value across all Cart types.

@dkotter Please take a look at this commit: [Don't clear coupon input if code doesn't exist](https://github.com/woocommerce/woocommerce/pull/48738/commits/11e7b8bf7441abe05715e5b09c10943536fe0afa)

**Review by dkotter — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**QA/Test Report-**  
------------

Thanks @Manussakis for checking the above concerns.


> Point-1
This is not an issue introduced by this PR. You can reproduce it on the trunk branch by:

ℹ️ Reported here  https://github.com/woocommerce/woocommerce/issues/50053   as a separate issue to address separately. 

> Point-2
Good catch! Let's keep consistency and preserve the value across all Cart types.

Point-2:  ✅ This issue has been fixed and works as expected now. ✅ 


Tested all scenarios on: 

- Tested on both shortcode-based cart/checkout pages and block-based cart/checkout.
- Tested with TT-4 theme, TT-1 theme, and Storefront. 

----


**Next Step:** Ready for code review.

**Comment by Manussakis** [comment]
@vikrampm1 This PR is ready to be reviewed by the appropriate Woo team.

**Review by nielslange — DISMISSED** [review]
Thanks for working on this, @Manussakis. While testing the classic cart against various themes, I noticed an inconsistency in how we display the error message when an invalid coupon code had been provided.

## Twenty Twenty-Four 

✅ This is what I expect to see for all themes.

Screenshot 2024-07-29 at 12 07 24

## Twenty Twenty-Three

❌ I expect the error message in red and below the coupon input field.

Screenshot 2024-07-29 at 12 08 40

## Twenty Twenty-Two

❌ I expect the error message in red and below the coupon input field.

Screenshot 2024-07-29 at 12 10 09

## Twenty Twenty-One

❌ I expect the error message in red, left-aligned with the coupon input field, and with more spacing between the input field and the error message.

Screenshot 2024-07-29 at 12 10 36

## Twenty Twenty

❌ I expect the error message in red, left-aligned with the coupon input field, and with more spacing between the input field and the error message.

Screenshot 2024-07-29 at 12 12 12

## Storefront

❌ I expect the error message in red, left-aligned with the coupon input field, and with more spacing between the input field and the error message.

Screenshot 2024-07-29 at 12 12 48

IMO, the error code should always appear left-aligned, below the input field, and displayed in red. Preferably, the input field should also have a red border when the coupon code is invalid.

@nikkivias What are your thoughts regarding the styling of the error messages, which I mentioned above?

---

I also noticed that when clicking into a different browser window, the focus gets lost. While the test steps state _"Verify the error message disappears after moving the focus away from the input [...]"_, is it expected that the focus gets lost when leaving the browser, e.g. to double-check the coupon code in another browser window or another app, e.g. an email client?

**Comment by Manussakis** [comment]
@nielslange Thanks for testing this PR.
The reason the error message style is inconsistent across different themes is that not all themes load the WooCommerce default CSS files. Only the Twenty Twenty-Four one does it and that's why it works as expected in this case.
This PR introduces a new CSS class named `coupon-error-notice` and themes don't have their own CSS rules for it yet.
I am curious about how would I fix this issue since we don't have control over whether themes will use WooCommerce default styles or not. Could you give more guidance on that?

-----

> I also noticed that when clicking into a different browser window, the focus gets lost [...]

I tested it again and couldn't reproduce the loss of focus. What I did notice is that the error message disappeared, but when I got back to the page, the coupon input still had the focus. Could you give more context on how you are having this issue?

**Comment by Manussakis** [comment]
Hey @nielslange 
I am just following up to see if you have any feedback on my last comment.
Thanks!

**Comment by vikrampm1** [comment]
@nielslange this is currently blocked and awaiting your feedback.

**Comment by mikejolley** [comment]
> This PR introduces a new CSS class named coupon-error-notice and themes don't have their own CSS rules for it yet.
I am curious about how would I fix this issue since we don't have control over whether themes will use WooCommerce default styles or not. Could you give more guidance on that?

We should at least add the styling rules for default WP themes to cover this. You can see by searching `.twenty` we have styles in place for those. These default themes should also load default Woo stylesheets? Only woo custom themes like Storefront would disable these.

**Comment by Manussakis** [comment]
@mikejolley Thanks for the feedback!

@dkotter Please re-review this task.

File: plugins/woocommerce/client/legacy/js/frontend/cart.js [hunk 5]
```diff
@@ -525,9 +584,21 @@ jQuery( function ( $ ) {
...
 				success: function ( response ) {
 					$(
-						'.woocommerce-error, .woocommerce-message, .woocommerce-info, .is-error, .is-info, .is-success'
+						'.woocommerce-error, .woocommerce-message, .woocommerce-info, .is-error, .is-info, .is-success, .coupon-error-notice'
 					).remove();
-					show_notice( response );
+					
+					// We only want to show coupon notices if they are not errors.
+					// Coupon errors are shown under the input.
+					if ( response.indexOf('woocommerce-error') === -1 && response.indexOf('is-error') === -1 ) {
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/cart.js]:
```suggestion
					if ( response.indexOf( 'woocommerce-error' ) === -1 && response.indexOf( 'is-error' ) === -1 ) {
```

File: plugins/woocommerce/client/legacy/js/frontend/checkout.js [hunk 6]
```diff
@@ -647,13 +676,20 @@ jQuery( function( $ ) {
...
 					$( '.woocommerce-error, .woocommerce-message, .is-error, .is-success' ).remove();
 					$form.removeClass( 'processing' ).unblock();
 
-					if ( code ) {
-						$form.before( code );
-						$form.slideUp();
+					if ( response ) {
+						// We only want to show coupon notices if they are not errors.
+						// Coupon errors are shown under the input.
+						if ( response.indexOf('woocommerce-error') === -1 && response.indexOf('is-error') === -1 ) {
```
Comment (dkotter) [inline plugins/woocommerce/client/legacy/js/frontend/checkout.js]:
```suggestion
						if ( response.indexOf( 'woocommerce-error' ) === -1 && response.indexOf( 'is-error' ) === -1 ) {
```

**Review by dkotter — APPROVED** [review]
@ankitguptaindia This is ready for another round of QA

**Comment by ankitguptaindia** [comment]
Hello @Manussakis, 


Tested with WordPress Default themes and Woo Storefront.

- Twenty Nineteen
- Twenty Twenty
- Twenty Twenty-One
- Twenty Twenty-Two
- Twenty Twenty-Three
- Twenty Twenty-Four



-  Classic Cart and Checkout Pages and,
-  Storefront -  Block and Classic Cart/Checkout pages.
- Twenty-twenty-four - Block and Classic Cart/Checkout pages. 


I have a few points to confirm:

1. In WordPress default themes with Classic Cart & Checkout pages, when a user adds an invalid coupon code, the field is cleared on the Cart page but remains populated on the Checkout page. Can we ensure consistent behavior for coupon fields on both pages? On the Block Cart/Checkout pages, the invalid coupon code remains in the field on both pages after the error notice.

<Details>
  <summary>Click for more details</summary>



https://github.com/user-attachments/assets/8e4942a7-e85f-42b9-a8d0-45692115a517

</Details>

2.  With Classic Checkout, when an invalid coupon code message appears, the border color of the cart field looks correct, but on the checkout page, it appears differently, as shown in the screenshot below.

<Details>
  <summary>Click for more details</summary>



| Cart Page | Checkout Page |
|--------|--------|
| image | image | 

</Details>

3. A question regarding the Storefront theme: 

>Note: I noticed that when the Storefront theme is installed, these CSS files aren't loaded on the classic Cart and Checkout pages:

On the Storefront theme, the validation message appears differently, particularly in terms of error message color and alignment (the reason is already mentioned in the description). Does this require any fixes in the theme repository? Since Storefront is a WooCommerce-supported theme, it would be great if it worked efficiently with it. 


<Details>
  <summary>Click for more details</summary>



image





</Details>

**Comment by Manussakis** [comment]
Hey @ankitguptaindia 

Replying to your points:

**Point 1**
> In WordPress default themes with Classic Cart & Checkout pages, when a user adds an invalid coupon code, the field is cleared on the Cart page but remains populated on the Checkout page. [...]

Thanks for flagging that. Fixed on [Keep input with the invalid coupon code after notice appears](https://github.com/woocommerce/woocommerce/pull/48738/commits/d2ffffd4e3d4c0657c9164888a222be21147fccb)

@dkotter please review this commit.

**Point 2**

> With Classic Checkout, when an invalid coupon code message appears, the border color of the cart field looks correct, but on the checkout page, it appears differently, as shown in the screenshot below.

The coupon field uses the same input style as the other ones on the checkout page, so it's consistent with them.
Screenshot at Aug 13 16-05-31

**Point 3**
[As recommended](https://github.com/woocommerce/woocommerce/pull/48738#issuecomment-2273759321) by @mikejolley I have provided styles for the default WP themes. My understanding is that we can't cover it for the Storefront theme since it doesn't load the default Woo stylesheet.

**Review by dkotter — APPROVED** [review]

**Comment by Manussakis** [comment]
@ankitguptaindia Please check if point-1, flagged by you, is now working as expected. Thanks!

**Comment by ankitguptaindia** [comment]
Thanks for the fixes @Manussakis Changes look great to me. 

PR is ready for the next step.  
cc: @vikrampm1

**Comment by vikrampm1** [comment]
@nielslange @mikejolley this is back for your re-review!

File: plugins/woocommerce/client/legacy/js/frontend/checkout.js [hunk 7]
```diff
@@ -647,13 +676,20 @@ jQuery( function( $ ) {
...
-				success:	function( code ) {
+				success:	function( response ) {
 					$( '.woocommerce-error, .woocommerce-message, .is-error, .is-success' ).remove();
 					$form.removeClass( 'processing' ).unblock();
 
-					if ( code ) {
-						$form.before( code );
-						$form.slideUp();
+					if ( response ) {
+						// We only want to show coupon notices if they are not errors.
```
Comment (mikejolley) [inline plugins/woocommerce/client/legacy/js/frontend/checkout.js]:
```suggestion
						// We only want to show coupon notices if they are no errors.
```

File: plugins/woocommerce/client/legacy/js/frontend/cart.js [hunk 8]
```diff
@@ -176,6 +194,43 @@ jQuery( function ( $ ) {
 		$target.prepend( html_element );
 	};
 
+	/**
+	 * Shows coupon form errors.
+	 *
+	 * @param {string|object} html_element The HTML string response after applying an invalid coupon or a jQuey element.
```
Comment (mikejolley) [inline plugins/woocommerce/client/legacy/js/frontend/cart.js]:
```suggestion
	 * @param {string|object} html_element The HTML string response after applying an invalid coupon or a jQuery element.
```

**Review by mikejolley — CHANGES_REQUESTED** [review]
I tested a selection of themes and both checkouts/carts and it works now good job.

Please fix the typos I highlighted and check the CI here (https://github.com/woocommerce/woocommerce/actions/runs/10375805397/job/28726542138?pr=48738). Once the linting issues are solved we can merge.

**Comment by Manussakis** [comment]
@mikejolley Thanks for testing the PR.
I fixed the CI issues with these two commits:
- [Break notice message into two lines](https://github.com/woocommerce/woocommerce/pull/48738/commits/6960fc06d7d368469a2da40287049352a42107f1)
- [Break if statement into two lines](https://github.com/woocommerce/woocommerce/pull/48738/commits/001a270e57de9c6570c3bb0674e4c809cd333e4d)

cc: @dkotter

**Review by mikejolley — APPROVED** [review]

**Pull request merged**